### PR TITLE
Add subsection for external computing devices

### PR DESCRIPTION
**Pull request by warp1337**
This is my first draft as discussed in #386.

### CODE REVIEW

File: general_rules/ExternalComputing.tex [hunk 1]
```diff
@@ -26,6 +26,34 @@ \section{External computing}\label{rule:robot_external_computing}
 
 \textbf{Remark:} Teams are allowed to use their own software in the external computing devices (not only cloud services). This software must be publicly available to other teams for scientific purposes (evaluation, test, and benchmarking), as well as for TC for inspection. Although open-sourcing the software is not mandatory, this practice is advised and encouraged by the league.
 
+\subsection{External Computing Devices on site for SSPL (Hardware)}
```
Comment (kyordhel) [inline general_rules/ExternalComputing.tex]: Let's generalize to all leagues. This subsection is then renamed to:
*On-site External Computing Devices*

File: general_rules/ExternalComputing.tex [hunk 2]
```diff
@@ -26,6 +26,34 @@ \section{External computing}\label{rule:robot_external_computing}
...
+\label{rule:robot_external_computing_devices_sspl}
+
+\begin{itemize}
+  \item \textbf{Location:} all external computing devices, approved by the \iaterm{Technical Committee}{TC}
```
Comment (kyordhel) [inline general_rules/ExternalComputing.tex]: Style changes:
- All external (capitalize A)

File: general_rules/ExternalComputing.tex [hunk 3]
```diff
@@ -26,6 +26,34 @@ \section{External computing}\label{rule:robot_external_computing}
...
+  		(see~\ref{rule:robot_external_computing}), must be placed at a designated location close to the arena.
+  		The location features a table (or otherwise suitable furnishing) and an ethernet switch (wired)
+        that connects to the arena WIFI. In the remainder of this document, this location is called
+  		\textbf{E}xternal \textbf{C}omputing \textbf{R}esource \textbf{A}rea (ECRA).
```
Comment (kyordhel) [inline general_rules/ExternalComputing.tex]: ECRA? Nice! I would rephrase paragraph to:

*must be placed at **the** designated location (e.g. a table close to the arena), referred hereinafter as \iaterm{\textbf{E}xternal \textbf{C}omputing \textbf{R}esource \textbf{A}rea}{ECRA}.*

Let's leave the Switch thing as another item:

- The OC will announce the ECRA during setup days.
- A switch connected to the Arena Wireless Network will be available to teams.

File: general_rules/ExternalComputing.tex [hunk 4]
```diff
@@ -26,6 +26,34 @@ \section{External computing}\label{rule:robot_external_computing}
...
+  \item \textbf{Procedure:} before a teams' test run, external comp. devices must be setup in time.
+  		Immediately after a teams' test slot, all equipment must be removed from the ECRA in order to give ensuing
+        teams adequate space and time to setup their devices. No team should ever occupy the ECRA continuously.
+  		As soon as the referee indicates the teams' test slot starts, team members are strictly forbidden to touch
+        their external comp. devices until the current test run is over. Breaking this ``hands-off'' rule
+        is penalized with 0 (zero) points for the corresponding test. Using external comp. devices during the test
```
Comment (kyordhel) [inline general_rules/ExternalComputing.tex]: Penalties must be also stated in the section of special bonuses and penalties

File: general_rules/ExternalComputing.tex [hunk 5]
```diff
@@ -26,6 +26,34 @@ \section{External computing}\label{rule:robot_external_computing}
...
+        from any other location than the ECRA will be penalized with \textbf{at least 0 (zero) points for the test}.
```
Comment (kyordhel) [inline general_rules/ExternalComputing.tex]: Penalties must be also stated in the section of special bonuses and penalties

File: general_rules/ExternalComputing.tex [hunk 6]
```diff
@@ -26,6 +26,34 @@ \section{External computing}\label{rule:robot_external_computing}
...
+\subsection{External Computing Devices on site for SSPL (Hardware)}
+\label{rule:robot_external_computing_devices_sspl}
+
+\begin{itemize}
+  \item \textbf{Location:} all external computing devices, approved by the \iaterm{Technical Committee}{TC}
+  		(see~\ref{rule:robot_external_computing}), must be placed at a designated location close to the arena.
+  		The location features a table (or otherwise suitable furnishing) and an ethernet switch (wired)
+        that connects to the arena WIFI. In the remainder of this document, this location is called
+  		\textbf{E}xternal \textbf{C}omputing \textbf{R}esource \textbf{A}rea (ECRA).
+  \item \textbf{Procedure:} before a teams' test run, external comp. devices must be setup in time.
```
Comment (kyordhel) [inline general_rules/ExternalComputing.tex]: I find the paragraph a little bit long. We must also avoid abbreviations like *comp.*. My two cents:
*There may be up to \textbf{two} teams in ECRA at any time, the testing team and the upcoming one. Immediately after the test, the team must remove all equipment from ECRA.*

File: general_rules/ExternalComputing.tex [hunk 7]
```diff
@@ -26,6 +26,34 @@ \section{External computing}\label{rule:robot_external_computing}
...
+  		Immediately after a teams' test slot, all equipment must be removed from the ECRA in order to give ensuing
+        teams adequate space and time to setup their devices. No team should ever occupy the ECRA continuously.
+  		As soon as the referee indicates the teams' test slot starts, team members are strictly forbidden to touch
+        their external comp. devices until the current test run is over. Breaking this ``hands-off'' rule
+        is penalized with 0 (zero) points for the corresponding test. Using external comp. devices during the test
+        from any other location than the ECRA will be penalized with \textbf{at least 0 (zero) points for the test}.
+  \item \textbf{Limitations:} external computing devices currently have no limitation concerning computation
```
Comment (kyordhel) [inline general_rules/ExternalComputing.tex]: Capitalize E

File: general_rules/ExternalComputing.tex [hunk 8]
```diff
@@ -26,6 +26,34 @@ \section{External computing}\label{rule:robot_external_computing}
...
+  		power and form factor. However, in order to maintain enough space and a	clear arrangement (e.g. wrt
+        inspectability for referees), stand-alone screens are \textbf{not allowed}. Mice and keyboards connected
+        to external computing devices are allowed.
+  \item \textbf{Recommendations:} show up in time and setup up your devices early enough. If your team is up last,
```
Comment (kyordhel) [inline general_rules/ExternalComputing.tex]: This item is unnecessary.

File: general_rules/ExternalComputing.tex [hunk 9]
```diff
@@ -26,6 +26,34 @@ \section{External computing}\label{rule:robot_external_computing}
...
+        do not setup first. Consider the ECRA a FIFO queue. There will be enough space at the ECRA to establish a
+        buffer (of teams), so if you show up in time and everyone removes their equipment the transition will be smooth.
+\end{itemize}
+
+The referees, Technical Committee, and Organizing Committee members may run random checks anytime during the competition
```
Comment (kyordhel) [inline general_rules/ExternalComputing.tex]: Add a \testbf{Remark:} at the beginning of the paragraph. A reference to the Fair Play rule is necessary as well as specifying that the team will be disqualified (is stronger that mentioning just a zero-point penalty) for the test, or even from the entire competition.

**Comment by warp1337** [comment]
Will include the proposed changes soon so that all OC/TC people can have a look at it.

File: general_rules/ExternalComputing.tex (hunk 6 above)
Comment (warp1337) [reply general_rules/ExternalComputing.tex]: Two teams is too little IMHO, let's make it three.

**Comment by warp1337** [comment]
@kyordhel Comments are incorporated. To whom it may concern ...

**Comment by LoyVanBeek** [comment]
It's on the back of my head to review somewhere this week. 
@awesomebytes @airglow any comments?

File: general_rules/ExternalComputing.tex [hunk 10]
```diff
@@ -26,6 +26,37 @@ \section{External computing}\label{rule:robot_external_computing}
...
+  \item \textbf{Location:} All external computing devices, approved by the \iaterm{Technical Committee}{TC}
+  		(see~\refsec{rule:robot_external_computing}), must be placed at the designated location, e.g., a table close
+        to the arena, referred hereinafter as the \iaterm{\textbf{E}xternal \textbf{C}omputing \textbf{R}esource
+        \textbf{A}rea} {ECRA}. The OC will announce the ECRA during setup days. A switch connected to the arena
+        wireless network will be available to teams.
+  \item \textbf{Procedure:} Before a teams' test run, external devices must be setup in time.
+  		Immediately after a teams' test slot, all equipment must be removed from the ECRA in order to give ensuing
+        teams adequate space and time to setup their devices. As soon as the referee indicates the teams' test slot
+        starts, team members are strictly forbidden to touch their external devices until the current test run is over.
+        Breaking this ``hands-off'' rule is penalized with -25 points for the corresponding test run. Also, not removing
```
Comment (kyordhel) [inline general_rules/ExternalComputing.tex]: Replace ``hands-off'' with \quotes{hands-off} or \enquote{hands-off} (If csquote package is not there, add it).

File: general_rules/ExternalComputing.tex [hunk 11]
```diff
@@ -26,6 +26,37 @@ \section{External computing}\label{rule:robot_external_computing}
...
+        equipment within a reasonable time frame is penalized with -25 points for the tests run.
+  \item \textbf{Limitations:} External computing devices currently have no limitation concerning computation
+  		power and form factor. However, in order to maintain enough space and a	clear arrangement (e.g. wrt
+        inspectability for referees), stand-alone screens are \textbf{not allowed}. Mice and keyboards connected
+        to external computing devices are allowed.
+  \item \textbf{Remark:} Please keep the fair play rule~\ref{rule:fairplay} in mind. The referees, technical committee,
```
Comment (kyordhel) [inline general_rules/ExternalComputing.tex]: Remark outside the itemize environment.

**Comment by warp1337** [comment]
All done. *bump*

**Comment by kyordhel** [comment]
Awesome. Thank you @warp1337.
Let's wait for approval from another TC.

**Comment by moriarty** [comment]
I'm not a TC. The changes look mostly great.

And I agree, last year, was a little tough with a lot of teams basically having full setups with monitors etc... at what should have just contained the external compute devices.
And, as OC- it was hard to see who had hands on their keyboards/mice and I did need to tell teams to back away, because there was such a pile of keyboards/mice/screens.

I just wanted to ask- how long do we intend for this process to be in place?
Teams already get backed-up bringing robots in and out of the arena, setting up etc... and now they will be bringing in and removing their external compute devices?

In the long term, future... it would be best/easier if it were possible to just get a rack mount shelf, with a lock... and just leave them there for the week. Teams put them there, don't really touch them, and don't expect to use them/have much access to them as developer PCs while at the competition.

Or, perhaps get a 16 port KVM switch for each and league keyboard mouse, with all devices locked under the table or in a rack. But I would suspect teams would fight for this during non-scheduled setup/practice time.

**Comment by warp1337** [comment]
bump @awesomebytes @airglow @justinhart

**Comment by LoyVanBeek** [comment]
I agree with @moriarty here that a set up where there is just a server quietly spinning in a corner or rack all week would be ideal. 
If that is not possible, then indeed the rules as suggested. Maybe the preference for such 'quietly spinner servers in a corner' could be expressed.

**Comment by warp1337** [comment]
Thx @LoyVanBeek for the review. Nevertheless, I dont like the idea if teams bringing a rack and having it connected to arena wifi the entrie week 24/7. Just my 2 cent.

**Comment by LoyVanBeek** [comment]
Good point. 

I think there should be a centralized place/rack people can put their 1U server in, but then again not many people will have those. But indeed, using WiFi all the time could be problematic.

**Comment by justinhart** [comment]
A 1U server could represent a significant investment over infrastructure that a lab already owns. It's true that teams are setting up computers, but for the most part, they're setting up systems that look like they've been ripped off of the desk in their lab. As such, it seems like we'd just be wedging out labs who already have gear.

Why not just say that all labs need to have their computers set up on the desk at the start of the round? I don't think that any of these computers are connecting via wi-fi. Just require them to hook up to the ethernet.

**Comment by warp1337** [comment]
@justinhart That's basically what I intended to express when I wrote the subsection. Maybe the current version is too fuzzy? Can someone please comment on the actual PR?

**Comment by airglow** [comment]
Short summary:
* We need to place a table close to the entrance but still visible from the inside of the arena.
* The table will be divided into two, let's say by red tape. One side current team, one side next team.
* Switches connected to the Arena Wifi. 
* We need someone who is observing this area regularly.

I would add an introducing sentence for this section, what the motivation is behind (Reduce the setup time, giving equal space for preparation of external hardware, keep the setup area clean). From the issue it's directly clear from the subsection not.

In general I see all these separate sections for each sub-league critical. Doesn't it also make sense to include it to the other Leagues? However this then may will affect the command stations that some teams are using for starting and inspecting during the tasks.

**Comment by warp1337** [comment]
@airglow Thanks for the comment. I will add a short introduction/explanation

@LoyVanBeek @kyordhel @justinhart 

What do you think about applying these rules to every league? Maybe we start by only applying them to SSPL/DSPL this year and see how this works?

**Comment by kyordhel** [comment]
I say: keep the rulebook as simple and plain as possible. We have a 100+ pages document nobody ever reads full. Rules need to be simpler.
I also say: for sake of simplicity, let's have the same rules for all 3 leagues unless strictly necessary.

File: general_rules/ExternalComputing.tex (hunk 9 above)
Comment (justinhart) [reply general_rules/ExternalComputing.tex]: I'm not sure that I like the idea of having the compute devices moving while people are in line. That could really slow up the line going through. Think of how long it could potentially take to start an external device, start all of the software, and so forth. I'd rather teams leave their gear hooked up and simply not communicate with the device and/or start their software until they're about to compete, but I think that moving the physical devices around could be chaotic.

File: general_rules/ExternalComputing.tex (hunk 1 above)
Comment (justinhart) [reply general_rules/ExternalComputing.tex]: Shouldn't the availability of software to other teams be after the competition? Otherwise, a team could potentially copy another team's software.

**Comment by justinhart** [comment]
I agree with @kyordhel. If there are dozens of league-specific variants of the rules, it could become difficult to make fair decisions when teams come into conflict in their interpretations of those rules.

**Comment by warp1337** [comment]
@kyordhel @justinhart Alright. Agree.

**Comment by LoyVanBeek** [comment]
You have my blessing, for what it matters. @kyordhel your review still says "requested changes", is the current PR OK now?

**Comment by kyordhel** [comment]
@LoyVanBeek I would need to check everything again this weekend. However, whether I approve changes or not is somehow irrelevant due to:
1. As @justinhart already pointed out, moving external devices can become a logistical nightmare and reduce team's performance during the test.
2. I'm the only one officially reviewing the PR while other TC members have comments but I'm not sure if they grant a go or a no-go.

So far, to me is OK (thanks for the blessing, Father), but I'm not clicking that approve button until I see another approval.

File: general_rules/ExternalComputing.tex (hunk 1 above)
Comment (kyordhel) [reply general_rules/ExternalComputing.tex]: No.

To breach the gap between Cloud Computing and External Computing, software resources (not source code) must be made available to all. If you have an outstanding ASR I should be able to try it also as is in your servers. We want to promote sharing and knowledge exchange in the league.

Any comments @LoyVanBeek? @balkce? @swachsmu? @dirkholz?

File: general_rules/ExternalComputing.tex (hunk 9 above)
Comment (kyordhel) [reply general_rules/ExternalComputing.tex]: Me neither, but having the devices of all 12 teams can also be problematic. Do you have in mind any alternative solution?

**Comment by warp1337** [comment]
@kyordhel Thx again for commenting, but again, concerning 1. in your last comment ... **NOT** moving equipment is an even bigger nightmare. IMHO, having a structured approach that causes trouble in the first iteration is better than having no approach at all that will make all hell break lose.

**Comment by kyordhel** [comment]
Don't get me wrong @warp1337, I agree that having the equipment there (outside a locked rack) might be more problematic than moving equipment competition-wise. My point was if the rest of the TC/OC agrees on transferring the nightmare to the logistic. **I'm not taking any responsibility regarding solutions for this issue**. This is solely TC to decide. My reasons:

1. As OC Chair, I barely have people to assist in the competition. Coordinate teams moving external devices is something the OC can't afford.
2. As referee and ex-TC I understand the problem. Cheaters gonna cheat and it looks terrible to have a bunch of hackers tweaking outside the team area.

In conclusion, I will follow the rules, but I'm not hanging myself with that rope.

**Comment by warp1337** [comment]
@kyordhel Yes, I understand that, totally. So, what are the other OC/TC members saying? If required, we can also have a skype call on this to speed up the process. For me, this topic is extremely important, almost as much as the network topic ;)

**Comment by awesomebytes** [comment]
Hello, excuse my lateness in the discussion. By reading the rules it makes me think the ideal setup for the external computer device is a beefy laptop. Our current equipment is an Alienware Tower, and I guess others will have something similar in their labs. We aren't in condition to change this hardware.

I mention it because what I understand from `all equipment must be removed from the ECRA` is that we'll need to move the tower in this case, which implies disconnecting it from power (and previously, not having it connected to power). If I'm misunderstanding, sorry!

I find this problematic as computers refusing to boot properly is not such a rare thing (specially with nvidia cards in them), which could affect the performance of teams. If there is a short time slot to setup the machine, and we need it to boot (and without a screen, as this rules say `stand-alone screens are not allowed`) we may not know something went wrong, or just take too long to notice. Also the startup time of whatever software we need to run affects this.

I would be fine with having enough power sockets to keep the machines plugged, and just disconnect the only ethernet cable available to connect to the arena wifi. Or something similar to that.

**Comment by warp1337** [comment]
@awesomebytes Thank you for the reply, really! Finally, someone actually participating in the league replied, yeah!

Now we are at the _core_ of the issue that I want to address and we have evidence that I was not sooo wrong with opening this discussion. 

I understand, you have a high-end desktop machine, which is hard to setup and subsequently to remove on site, especially without a screen so on and so forth. Got it! 

I also get that you most probably won't buy a high end laptop instead, since you already have a machine for your robot. Fair enough.

In our case, we use a high-end laptop. To be precise, we are using this one: "Dell Inspiron 15 7000 Gaming-Notebook". So, no worries from our side...

Let's boil this down ... again: we will be > 10 Pepper teams at Montreal. If everyone brings their desktop machines, including a screen and a keyboard, we will ~probably~ end up in the same situation as we've seen in Nagoya --- total chaos at the ECRA and:

**1)** Local bandwidth will be quickly exhausted because everyone leaves their comp. machines running and streaming all the time.
    Even if you have the good will and plan on shutting it down when not needed, people will forget that when it's crunch time, trust me on that one.
**2)** There simply won't be enough space.
**3)** All referees will see from the ECRA is a 'wall of screens'

I ~expect~ hope people will make a decision at some point (soon-ish) regarding this PR/Issue. 

I totally see the point of bringing a high end desktop, simply more power. If there is no limitation in the rulebook, we will probably also switch to a desktop machine (thinking competition-wise). 

However, I really don't like the idea of making this a 'who has the most TitanX GPUs in their machine' battle -- especially because it comes with all the downsides I mentioned in 1) 2) and 3). Finally, why buying a ~15K robot if you need to additionally buy a 3-4K machine in order to run it (this is the SSPL, remember).

My 2 cent.

@LoyVanBeek @kyordhel  @moriarty @justinhart 

@LoyVanBeek  @kyordhel @moriarty @justinhart

**Comment by kyordhel** [comment]
### Current rules
@awesomebytes please open an issue indicating where the current rules are ambiguous regarding the use of a laptop as only alternative for external computing. So fas there is no restriction on the type of external computers that can be used.

### Pull request
Replying to @awesomebytes and @warp1337. I'm OK with an hybrid approach. Consider the following:
- **Laptops:** To be set up one slot before the test and removed after the test.
- **Desktops and servers:** To be set up early in the morning and shut down at evening (we can-t be sure power won-t be cut). The rest of the time the equipment must remain chained to the table with no keyboard/mouse/monitor plugged in.
    - In case of bandwidth problems, policy is: cable is connected before the test and disconnected afterwards.

As referee I can say: If I see anyone nearby those servers during a test, that team is immediately disqualified in the most bad-ass possible way.

**Comment by LoyVanBeek** [comment]
100% agree with the change that @kyordhel proposes. 

I feel the concern about "who can cram in the most GPUs in a box", but the same would apply when using a cloud service. This is a box of Pandora opened when external computing was allowed. Restricting on the amount of T/GFlops or something is  not easy either. Perhaps limit to 1 GPU, but that's all besides this issue.

**Comment by kyordhel** [comment]
@LoyVanBeek I think no restriction shall be imposed in the external computing devices, specially since:

> _The RoboCup@Home tournaments are organized in independent test sets, which are used to benchmark the robots' abilities and performance in a realistic non-standardized home environment. _ [1]

So, as part of this _realism_, the arena network should be much like the one given by standard domestic internet provider. If your external devices require high-end infrastructure, you are doing something wrong and definitively is not @Home's problem.

Yet again, there is another rule specifying that robots must remain safely functional when no wireless is available.

[1] Wisspeintner, T., Van Der Zant, T., Iocchi, L. and Schiffer, S., 2009. RoboCup@ Home: Scientific competition and benchmarking for domestic service robots. Interaction Studies, 10(3), pp.392-42

**Comment by warp1337** [comment]
>Laptops: To be set up one slot before the test and removed after the test.
> Desktops and servers: To be set up early in the morning and shut down at evening (we can-t be sure power won-t be cut). The rest of the time the equipment must remain chained to the table with no keyboard/mouse/monitor plugged in.
> In case of bandwidth problems, policy is: cable is connected before the test and disconnected afterwards.

Thanks. I would agree 100% to that proposal

**Comment by moriarty** [comment]
I am OK with the hybrid approach from @kyordhel above.

> Now we are at the core of the issue that I want to address and we have evidence that I was not sooo wrong with opening this discussion.

> This is a box of Pandora opened when external computing was allowed.

Yes I remember reading the TDPs last year and seeing what they had listed as "External Computing" and was a little shocked... and then I went on a rant in #243.

> So, as part of this _realism_ ...

@kyordhel  was talking network bandwidth there, but I think in the future I agree with discussing some limits like @LoyVanBeek suggests for the External Compute, at least for _majority_ of the competition... Open Challenge you shouldn't be held back and can do whatever you want/can afford. But that's another issue for later.

**Comment by kyordhel** [comment]
@warp1337 With "at least" we might have the situation in which all team have their equipment there since the beginning of the test. That worsens when tests are run in cycle, they might not remove them at all!

**Comment by justinhart** [comment]
Is it really necessary that teams not fiddle with whatever server between rounds? I'm not sure what function it serves.

**Comment by kyordhel** [comment]
@justinhart Basically to prevent last-minute tweaks and adjustments that in many cultures are not considered as cheating but might still grant a slight disadvantage to the competing team.

**Comment by warp1337** [comment]
Like starting to stream the hell out of the their depth cam point clouds. As proposed in the network issue bandwidth limitation would mitigate that. 

I recently spoke to a colleague who has also long standing experience with the RoboCup ... He introduced a, at least for me, totally different point of view. I will elaborate on that later...

**Comment by justinhart** [comment]
I see. Yes, that would be really uncool, but wouldn't they be able to do
this remotely?

**Comment by kyordhel** [comment]
> I see. Yes, that would be really uncool, but wouldn't they be able to do this remotely?

Maybe but not easily. Let's remember the External Devices and the robot run on the Arena Network, one to which only authorized MAC Addresses can connect. This makes unlikely that any other computer can access the server while, at the same time, makes possible for both, robot and external device to run in the same ROS environment.

@warp1337 That's an important point. Although the Arena Network is separated from the testing network, let's not forget all wireless devices share the same medium. This means that any computer streaming 1080p@60fps pointclouds through any wireless network will negatively impact all wireless within 200m radius.

**Comment by warp1337** [comment]
Hi everybody!

In the following I will try to summarize the current **general** situation and propose a completely new approach / introduce a completely new point of view. I will try to keep this post as short as possible.

My original idea behind the ECRA, as proposed above, was:

- limit the 'jumble' before the task, i.e., people fiddling with other peoples' cables and whatnot
- provide insight for refs of what is happening during the task wrt 'people typing in stuff on their laptops during their task' aka 'I am just inspecting log files' 
- limit overall network traffic by not having 'server-like' machines running and streaming **all** the time

I came up with these ideas after the RC in Nagoya, where we experienced an excellent network setup btw. To be honest, these ideas are based on my latest experience in the SSPL (after almost 8 years in the OPL). I recently had a good conversation with my colleague @semeyerz who lead team ToBI with me and Sven W. in 2016 (Leipzig). When I told him the current situation he was, mildly put, ... shocked.

Here is why, and this is also my current feeling about this. 

In the past, bringing a robot to the arena for a test, actually really meant bringing a **robot** --- not a robot, two laptops, one server and 12 team members setting all this up. It was just: bring your robot, press a start button, watch the robot _not_ enter the arena, repeat. Being part of the SSPL myself and having now worked with the Pepper for almost 1 year now, I 'get' that external computing is almost inevitable with that specific robot (prove me wrong!). 

Moreover, .... hand on heart ... since we allow external cloud computing --- _note:_ this connects the robot to the **internet** --- we cannot really control whether someone is remote controlling the robot or labeling data manually or not. Don't get me wrong please, I have trust in sportsmanship and fair play and I don't allege people do that in general. It's just, we simply **cannot control this anymore**, moving laptops in-between test doesn't help here.

So here's my new proposal:

- server-like machines ---> have to be setup in a designated area before the event, have to be removed after the event. There's no fiddling during the tests. In my personal opinion, as said before, I would ban server-like machines at all, but that's a) not up to me and b) a completely different story. IMHO, the same goes for external (internet) cloud services, btw.
- bring your robot ---> **two** team members are allowed in the area formerly known as the ECRA and **one** laptop-like device to set, e.g., the final initial pose for AMCL, calibrate ASR or whatnot. **No crowd** in the ECRA, no setting up entire workspaces, no nothing :)  Laptop has to be removed after the test.

There are a few 'dependencies' in order to make this work:

- fair play, respect sportsmanship, and keep an eye on your fellow team members.
- limit the bandwidth per team/vlan (a server may now stream as much data as it wants, it'll only
affect the bandwidth within the vlan)

By implementing this, we will solve the above mentioned problems without introducing complicated rules about setting up / tearing down stuff.

What do you think of all this?

File: general_rules/ExternalComputing.tex (hunk 9 above)
Comment (justinhart) [reply general_rules/ExternalComputing.tex]: I'm not sure. The way this is stated really makes problems, though. There shouldn't be an external monitor, meaning that the external device either must be a laptop or permit someone to run SSH or a terminal emulator into the system. Plus you have boot times to think of. Consider how crazy it is when the ref is going, "<Team Name>, it's time to set up your robot and compete or forfeit your turn." With this setup, teams may still be booting up their hardware and wrestling with network problems during that time. I really think that we just need something like a long table, not inside the lane where the teams are going, and then a requirement that you unplug your network cable until it's your turn. This allows teams to get their computers going without much issue. Maybe we give them a max footprint for their hardware on the table. Everything must fit inside this square.

**Comment by justinhart** [comment]
I was just commenting something similar in Github. Sorry for being a bit
behind the stick. I had some personal matters come up. I can agree to this.

**Comment by warp1337** [comment]
Thx, @justinhart for the comment.

>  had some personal matters come up.

Hope you're fine?!

What about the others @airglow @awesomebytes @LoyVanBeek  @kyordhel

**Comment by awesomebytes** [comment]
I agree with the last view/proposal of @warp1337.

Letting the server-like devices in the place and not touching them. Also unplugging Ethernet cables sounds like a good way to ensure the competing team has all the available bandwidth.

Having just some specific people to access a laptop for the initial  setup of a test and get hands off it sounds reasonable too.

:+1:

**Comment by warp1337** [comment]
I feel like I need to clarify one further detail. My proposal does not solve the bandwidth issue. It's just a simple and clear proposal about the procedure concerning the tests. However, the bandwidth 'situation' can only be solved by the LOC, i.e., limiting the available bandwidth per team (vlan) based on the infrastructure's capabilities --- put simply: total available wifi bandwidth is 100 Mbps and number of teams is 10 ---> 100/10 per vlan.

**Comment by kyordhel** [comment]
I would like to request only three things:
- Servers must be chain-locked to the table (anti-theft).
- No people around them. Ever.
  - No plugging/unplugging cables (accidental sabotage?).
  - No bluetooth mouses, keyboards, tablets, etc. nrearby.
- SPL team members volunteer as cops
  - You see anything suspicious, you report.
  - No warnings, immediate disqualification.

**Comment by warp1337** [comment]
If more than one members of the committee agree to this, I am willing to rework my PR. Thx @kyordhel

**Comment by kyordhel** [comment]
@warp1337 

You have @justinhart and @awesomebytes explicit approval, as well as the tacit ones from the rest of the TC. Please proceed to make the changes in the PR.

P.S. My requests are on the operative level. Like the document distributed about Network configuration, detailed information shouldn't be in the rulebook, but standalone as specific regulations for this particular event.

**Comment by warp1337** [comment]
Okay. Will do. Will take some hours, thank you.

**Comment by warp1337** [comment]
I updated the PR, please review (sorry took more than a few hours...)

@LoyVanBeek @kyordhel @justinhart

**Comment by warp1337** [comment]
Thank you! Finally, we could all agree :)
I believe it is important to _notify_ the people about this addition. How can we do this?

**Comment by kyordhel** [comment]
@warp1337 Several changes (winter) are coming and must be committed/merged before Sunday. Then we will have an official draft which will be sent to all teams by mail (OC do this).
